src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -0,0 +1,170 @@
+  private MotorController hood_motorController;
+  private SparkMaxPIDController KShooterController;
+  private SparkMaxPIDController KHoodController;
+  private RelativeEncoder KShooterEncoder;
+  private RelativeEncoder KHoodEncoder;
+  private MotorController cargo_motorController;
+  private SparkMaxPIDController kCargoController;
+  private RelativeEncoder kCargoEncoder;
+
+  public ShooterSubsystem() {
+
+    aimMode = 1;
+    cargo_motorController = new MotorController("Shooter Cargo", Constants.KShooterCargoID);
+    kCargoController = cargo_motorController.getPID();
+    kCargoEncoder = cargo_motorController.getEncoder();
+
+    shooter_motorController = new MotorController("Shooter", Constants.KShooterID);
+    KShooterController = shooter_motorController.getPID();
+    KShooterEncoder = shooter_motorController.getEncoder();
+    hood_motorController = new MotorController("Hood", Constants.KHoodID);
+    KHoodController = hood_motorController.getPID();
+    KHoodEncoder = shooter_motorController.getEncoder();
+
+  }
+
+  public void adjustHood(double a) {
+    KHoodController.setReference(a, CANSparkMax.ControlType.kPosition);
+
+    // Adjusts Hood using PID control to passed angle a
+  }
+
+  public void windFlywheel(int rpm) {
+
+    double kGearRationRPM = rpm * (Constants.KGearRatioIn / Constants.KGearRatioOut);
+    KShooterController.setReference(kGearRationRPM, CANSparkMax.ControlType.kVelocity);
+    while (true) {
+
+      double KFlywheelspeed = KShooterEncoder.getVelocity();
+      if (KFlywheelspeed == rpm) {
+        // Run the last wheel
+        kCargoController.setReference(Constants.kCargoRotation, CANSparkMax.ControlType.kPosition);
+
+        break;
+      }
+
+    }
+
+    // Winds Flywheel using PID control to passed rpm
+  }
+
+  public void setAimMode(int m) {
+    aimMode = m;
+  }
+
+  public void cycleAimModeUp() {
+    aimMode++;
+    if (aimMode > 3) {
+      aimMode = 0;
+    }
+  }
+
+  public void cycleAimModeDown() {
+    aimMode--;
+    if (aimMode < 0) {
+      aimMode = 3;
+    }
+  }
+
+  public double getTY() {
+    // Gets TY, the vertical angle of the target from the limelight
+    return Math.toDegrees(NetworkTableInstance.getDefault().getTable("limelight").getEntry("ty").getDouble(0));
+  }
+
+  public double getDistance() {
+    // Uses Limelight to find distance to High Goal
+    return (Constants.kGoalHeight - Constants.kLLHeight) / Math.tan(getTY() + Constants.kLLAngle); // Return distance in
+                                                                                                   // feet
+  }
+
+  public double[] ProjectilePrediction(double y0, double x0, double y, double x, double g, double t) {
+    /**
+     * Fangle =math.degrees(math.atan((y-y0+1/2*g*(t**2))/x))
+     * velcocity1 = abs(x/(math.cos(math.radians(Fangle))*t))
+     * velcocity2 = abs((y-y0+(g/2.0)*(t**2))/(math.sin(math.radians(Fangle))*t))
+     * return Fangle, velcocity1, velcocity2
+     */
+    x = x + 1;
```

Comment:
Is this to prevent non-zero x?  Would it be better to just return an error if there is a zero x?

Reply:
This is an offset applied to x so that we are shooting to the center of our goal, rather than the rims, where the vision tape is.